src/core/adapters/search.ts

```diff
@@ -144,10 +144,26 @@ namespace Adapter {
     };
   };
 
-  export const extractProducts = (results: Results) =>
-    results.records.map(Adapter.extractProduct);
+  export const extractData = (products: Store.ProductWithMetadata[]) =>
+    products.map(({ data }) => data);
 
-  export const extractProduct = ({ allMeta }) => allMeta;
+  export const extractProducts = (state: Store.State, results: Results) => {
+    const pastPurchases = Selectors.pastPurchaseProductsBySku(state);
+    let startIndex = results.pageInfo.recordStart;
+    return results.records.map(({ allMeta }) => {
+      const meta: any = {};
+
+      if (allMeta[state.session.config.recommendations.idField] in pastPurchases) {
```

Comment:
we should only add this when selecting the data back out to ensure we never get out of sync